### PR TITLE
feat/main layout/ users search

### PR DESCRIPTION
**Pull request by vlzelensky**

### CODE REVIEW

File: src/components/Layout/Layout.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import './styles.css';
+
+const { Header, Sider, Content } = AntdLayout;
+
+export const Layout: FC<LayoutProps> = observer(({ children }) => {
+  const [collapsed, setCollapsed] = useState<boolean>(true);
+  const navigate = useNavigate();
+
+  const avatarText = user.userName
```
Comment (dariausatova) [inline src/components/Layout/Layout.tsx]: destructurization user

File: src/components/Layout/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  return (
+    <AntdLayout className='layout'>
+      <Header className='header'>
+        <img className='logo' src={logo} alt='logo' />
+        <Search />
+        <div>
+          <span className='user-name' onClick={() => navigate('/profile')}>
+            {user.userName}
+          </span>
+          <Button className='header_button' onClick={() => user.logout()}>
```
Comment (dariausatova) [inline src/components/Layout/Layout.tsx]: stick to the same style of writing class names. a kebab-case is always used

File: src/components/Layout/Layout.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        </div>
+      </Header>
+      <AntdLayout>
+        <Sider
+          collapsible
+          collapsed={collapsed}
+          onCollapse={(value) => setCollapsed(value)}
+        >
+          <div className='sider-container'>
+            <div style={{ flex: 1 }}>
```
Comment (dariausatova) [inline src/components/Layout/Layout.tsx]: what is it? why are styles described like this? do you have files where you describe styles

File: src/components/Layout/styles.css [hunk 4]
```diff
@@ -0,0 +1,45 @@
+.layout {
```
Comment (dariausatova) [inline src/components/Layout/styles.css]: why isn't scss used?

File: src/components/Search/Search.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import { useDebounce } from 'hooks';
+import { findUser } from 'api';
+import { user } from 'store';
+
+import './styles.css';
+
+export const Search: FC = () => {
+  const [inputValue, setInputValue] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [users, setUsers] = useState<null | { id: string; userName: string }[]>(null);
```
Comment (dariausatova) [inline src/components/Search/Search.tsx]: create user interface

File: src/components/Search/Search.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    }
+  }, [debouncedSearch]);
+
+  return (
+    <div className='search-wrapper'>
+      <AutoComplete
+        className='search'
+        options={users?.map((user) => ({ value: user.userName, ...user }))}
+        value={inputValue}
+        onChange={(value) => setInputValue(value)}
```
Comment (dariausatova) [inline src/components/Search/Search.tsx]: why is it not typed?

File: src/components/Search/Search.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        setUsers(null);
+      }
+    }
+  }, [debouncedSearch]);
+
+  return (
+    <div className='search-wrapper'>
+      <AutoComplete
+        className='search'
+        options={users?.map((user) => ({ value: user.userName, ...user }))}
```
Comment (dariausatova) [inline src/components/Search/Search.tsx]: why is it not typed?

File: src/hooks/useDebounce.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { useState, useEffect } from 'react';
+
+export const useDebounce = (value: string, delay: number) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  useEffect(() => {
```
Comment (dariausatova) [inline src/hooks/useDebounce.ts:6]: add empty line

File: src/store/user.ts [hunk 9]
```diff
@@ -1,54 +1,67 @@
...
+import { logIn, getUser, findUser } from 'api';
 import { initApi } from 'config/axios';
+import { LoginResponse } from 'api/types';
 
 class User {
   constructor() {
     makeAutoObservable(this);
   }
 
+  id: string | null = null;
```
Comment (dariausatova) [inline src/store/user.ts]: why do I need to store all the fields separately? why not store them as an object?

File: src/store/user.ts [hunk 10]
```diff
@@ -1,54 +1,67 @@
...
   isAuth: boolean = false;
```
Comment (dariausatova) [inline src/store/user.ts]: such variables are declared in the constructor and via this

File: src/store/user.ts [hunk 11]
```diff
@@ -1,54 +1,67 @@
...
-        this.birthDate = null;
+        this.logout();
       }
     }
   }
+
+  async searchUser(value: string, userId: string) {
+    try {
+      const res = await findUser(value, userId);
+    } catch (e) {}
```
Comment (dariausatova) [inline src/store/user.ts]: why is catch not described?

File: src/store/user.ts [hunk 12]
```diff
@@ -1,54 +1,67 @@
...
-        this.userName = null;
-        this.birthDate = null;
+        this.logout();
       }
     }
   }
+
+  async searchUser(value: string, userId: string) {
+    try {
+      const res = await findUser(value, userId);
```
Comment (dariausatova) [inline src/store/user.ts]: you don't need to return anything from the function?

File: src/store/user.ts (hunk 10 above)
Comment (vlzelensky) [reply src/store/user.ts]: ts will compile it as:
 constructor() {
        this.id = null;
    }
   anyway

File: src/components/Layout/Layout.tsx [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+  return (
+    <AntdLayout className='layout'>
+      <Header className='header'>
+        <img className='logo' src={logo} alt='logo' />
+        <div>
+          <span className='user-name' onClick={() => navigate('/profile')}>
+            {name}
+          </span>
+          <Button className='header-button' onClick={() => user.logout()}>
```
Comment (dariausatova) [inline src/components/Layout/Layout.tsx:34]: destructurization user higher

File: src/components/Layout/styles.css [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
```
Comment (dariausatova) [inline src/components/Layout/styles.css]: I don't see a scss connection

File: src/components/Search/Search.tsx [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+export const Search: FC = () => {
+  const [inputValue, setInputValue] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [users, setUsers] = useState<null | { id: string; userName: string }[]>(null);
+  const debouncedSearch = useDebounce(inputValue, 1000);
+  const navigate = useNavigate();
+
+  const openConversation = (value: string) => {
+    const user = users?.find((el) => el.userName === value);
+    navigate(`/conversation/${user?.id}`);
```
Comment (dariausatova) [inline src/components/Search/Search.tsx]: id is not a required field? or do I need to check first that the user is there, and only then make the transition?

File: src/components/Search/Search.tsx [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+  };
+
+  useEffect(() => {
+    if (debouncedSearch && inputValue.length > 2) {
+      setIsLoading(true);
+      findUser(inputValue, user.id!).then((res) => {
```
Comment (dariausatova) [inline src/components/Search/Search.tsx]: is it impossible to destruct res?

File: src/components/Search/Search.tsx [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+        setUsers(null);
+      }
+    }
+  }, [debouncedSearch]);
+
+  return (
+    <div className='search-wrapper'>
+      <AutoComplete
+        className='search'
+        options={users?.map((user: { id: string; userName: string }) => ({
```
Comment (dariausatova) [inline src/components/Search/Search.tsx]: interface for user

File: src/store/user.ts (hunk 10 above)
Comment (dariausatova) [reply src/store/user.ts]: without this prescribe

File: src/components/Layout/styles.scss [hunk 18]
```diff
@@ -0,0 +1,49 @@
+.layout {
```
Comment (dariausatova) [inline src/components/Layout/styles.scss:1]: and where did the principles of sccs apply?

**Pull request merged**